### PR TITLE
Minor script fixes

### PR DESCRIPTION
**Pull request by noahyor**
Replaces the old (quite ugly and unreadable) code for finding the swarm root dir with
`git rev-parse --show-toplevel`

### CODE REVIEW

File: scripts/gen/preview-world-vscode.sh [hunk 1]
```diff
@@ -17,21 +17,14 @@
 # preview with CTRL+PageUp, and then
 # CTRL+w will close the redundant image preview.
 
-SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-cd $SCRIPT_DIR/..
+cd $(git rev-parse --show-toplevel)
```
Comment (byorgey) [inline scripts/gen/preview-world-vscode.sh:20]: Are you sure this is the correct directory?  Above had `$SCRIPT_DIR/..` .  Though it's possible that one became incorrect when this script was moved.

File: scripts/gen/preview-world-vscode.sh [hunk 2]
```diff
@@ -17,21 +17,14 @@
...
-
-IMG_OUTPUT_PATH=output.png
-RENDER_IMG_COMMAND="stack exec swarm-scene -- $SCENARIO_PATH --fail-blank --dest $IMG_OUTPUT_PATH --png --width $IMG_WIDTH --height $IMG_HEIGHT"
-
 stack build --fast swarm:swarm-scene
 
-$RENDER_IMG_COMMAND
-code --reuse-window $SCENARIO_PATH && code --reuse-window $IMG_OUTPUT_PATH 
+stack exec swarm-scene -- $SCENARIO_PATH --fail-blank --dest outpup.png --png --width 200 --height 150
+code --reuse-window $SCENARIO_PATH && code --reuse-window outpup.png
```
Comment (byorgey) [inline scripts/gen/preview-world-vscode.sh:28]: What is the purpose of inlining all these definitions?  I find this version less readable than the original version that defined variables like `IMG_WIDTH` and so on.  Also, it should be `output.png` not `outpup.png`.

File: scripts/gen/render-sublibrary-dependencies.sh [hunk 3]
```diff
@@ -1,7 +1,6 @@
 #!/bin/bash -ex
 
-SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-cd $SCRIPT_DIR/../..
+cd $(git rev-parse --show-toplevel)/..
```
Comment (byorgey) [inline scripts/gen/render-sublibrary-dependencies.sh:3]: Should this really have `/..` after the `git rev-parse ...` command?

File: scripts/gen/schema-docs.sh [hunk 4]
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash -e
 
-SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-cd $SCRIPT_DIR/../..
+cd $(git rev-parse --show-toplevel)/..
```
Comment (byorgey) [inline scripts/gen/schema-docs.sh:3]: This also has `/..` which I'm not sure should be there.

File: scripts/gen/preview-world-vscode.sh (hunk 1 above)
Comment (noahyor) [reply scripts/gen/preview-world-vscode.sh:20]: I think this was just never fixed from when we made subdirs.

File: scripts/normalize/yaml.sh [hunk 5]
```diff
@@ -1,7 +1,6 @@
 #!/bin/bash -e
 
-SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-cd $SCRIPT_DIR/../..
+cd $(git rev-parse --show-toplevel)/..
```
Comment (byorgey) [inline scripts/normalize/yaml.sh:3]: Missed this one?

File: scripts/validate/issues-for-todos.sh [hunk 6]
```diff
@@ -1,7 +1,6 @@
 #!/bin/bash -e
 
-SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-cd $SCRIPT_DIR/../..
+cd $(git rev-parse --show-toplevel)/..
```
Comment (byorgey) [inline scripts/validate/issues-for-todos.sh:3]: And this one.

File: scripts/validate/json-schemas.sh [hunk 7]
```diff
@@ -1,7 +1,6 @@
 #!/bin/bash -e
 
-SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-cd $SCRIPT_DIR/../..
+cd $(git rev-parse --show-toplevel)/..
```
Comment (byorgey) [inline scripts/validate/json-schemas.sh:3]: One more.

**Review by byorgey — COMMENTED** [review]
Looks good to me, just a few stray `..` to remove still.

File: scripts/gen/preview-world-vscode.sh (hunk 1 above)
Comment (byorgey) [reply scripts/gen/preview-world-vscode.sh:20]: I think you're right.

**Review by byorgey — APPROVED** [review]

**Review by xsebek — APPROVED** [review]
Neat! 👍

**Pull request merged**